app/ssttpcalculator/model/PaymentPlanOption.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ssttpcalculator.model
+
+import enumeratum._
+
+import scala.collection.immutable
+
+sealed trait PaymentPlanOption extends EnumEntry with EnumEntry.Uncapitalised {
+  val idKey: String
+}
+
+object PaymentPlanOption extends Enum[PaymentPlanOption] with PlayInsensitiveJsonEnum[PaymentPlanOption] {
+  val values: immutable.IndexedSeq[PaymentPlanOption] = findValues
+
+  case object Basic extends PaymentPlanOption {
+    val idKey = "basic"
```

Comment:
I think these `idKey` values are the same ones that you get with `.entryName` from `EnumEntry` - is it possible to use `entryName` instead of `idKey` if it is? Or did you specifically want something called `idKey`? If you did then it might be an option to have:
```scala
sealed trait PaymentPlanOption extends EnumEntry with EnumEntry.Uncapitalised {
  val idKey: String = entryName
}
```
to save having to define the values in the case objects.